### PR TITLE
Feature/list catalogue items

### PR DESCRIPTION
**Pull request by crypt0s1s**
Mitch to write description

### CODE REVIEW

File: Helios/api/store/shop/store.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { flow, types } from "mobx-state-tree";
+import { authHeaderInterceptor, createHeliosApi } from "../../core";
+import { ShopItemModel } from "./models";
+import { useQuery } from "@tanstack/react-query";
+import { rootStore } from "@/api";
+// import { ClientTokenResponse } from "./models";
```
Comment (crypt0s1s) [inline Helios/api/store/shop/store.ts:6]: commented code

File: Helios/api/store/shop/models/shopItemModel.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { types, Instance } from "mobx-state-tree";
+
+export const ShopItemModel = types.model({
+  id: types.identifier,
+  name: types.string,
+  minPrice: types.float,
+  maxPrice: types.maybeNull(types.float),
```
Comment (crypt0s1s) [inline Helios/api/store/shop/models/shopItemModel.ts:7]: lets remove this for now. Comment it out. I'll look into a fix for this later. Add a `// TODO: Josh to fix up`

File: Helios/api/store/shop/store.ts (hunk 1 above)
Comment (MitchMunro) [reply Helios/api/store/shop/store.ts:6]: Are you just noting that there is commented code when there shouldn't be?

Comment (crypt0s1s) [reply Helios/api/store/shop/store.ts:6]: yes. It should be deleted unless it has a reason to exist.

File: Helios/api/store/shop/store.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+export const ShopStore = types
+  .model("ShopStore", {
+    shopItems: types.array(ShopItemModel),
+  })
+  .actions((self) => ({
+    getShopItems: flow(function* getShopItems() {
+      let result = yield shopApi.get("").json();
+      console.log(result);
+      self.shopItems = result;
+      console.log(self.shopItems);
```
Comment (crypt0s1s) [inline Helios/api/store/shop/store.ts:24]: lets remove the console.log actions

File: Helios/api/store/shop/store.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+export const shopApi = createHeliosApi(shopUrl);
+
+export const authenticatedShopApi = createHeliosApi(shopUrl, [
+  authHeaderInterceptor,
+]);
+
+export const ShopStore = types
+  .model("ShopStore", {
+    shopItems: types.array(ShopItemModel),
+    shopItemDetails: types.array(ShopItemDetailsModel),
```
Comment (crypt0s1s) [inline Helios/api/store/shop/store.ts:19]: this shouldn't be an array

File: Helios/api/store/shop/store.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  })
+  .actions((self) => ({
+    getShopItems: flow(function* getShopItems() {
+      let result = yield shopApi.get("").json();
+      console.log("Test result 1: " + result);
```
Comment (crypt0s1s) [inline Helios/api/store/shop/store.ts:24]: can we remove the console.log's

File: Helios/api/store/shop/store.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+export const getShop = () => {
+  return useQuery({
+    queryKey: ["shop"],
+    queryFn: rootStore.shop.getShopItems,
+  });
+};
+
+export const getShopDetails = () => {
+  return useQuery({
+    queryKey: ["shopDetails"],
```
Comment (crypt0s1s) [inline Helios/api/store/shop/store.ts:46]: Let's update the query key to `["shop", id]` instead. This value is used f or caching. This function should also take the item id that it's fetching for. Have a read of the docs about query keys when you get a chance. https://tanstack.com/query/v5/docs/framework/react/guides/query-keys

File: Helios/pages/mitchTest/[id].tsx
Comment (crypt0s1s) [inline Helios/pages/mitchTest/[id].tsx:1]: let's not commit code like this. You can have it locally but no need to commit it.

File: Helios/pages/mitchTest/index.tsx
Comment (crypt0s1s) [inline Helios/pages/mitchTest/index.tsx:1]: same with this

File: Helios/pages/shop/SearchPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      </div>
+      <div className="text-lg">
+        <b>Search Items</b>
+      </div>
+      <SearchBar />
+    </div>
+  );
+};
+
+function SearchBar() {
```
Comment (crypt0s1s) [inline Helios/pages/shop/SearchPanel.tsx:18]: We should be creating base components like these in Helena. I'll be spending some time building that up soonish.

File: Helios/pages/shop/[id].tsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+const Page: NextPageWithLayout = () => {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const { shop } = useContext(StoreContext);
+  const { isPending, isSuccess, error } = getShopDetails();
+  console.log("Shop Item error: " + error);
+  console.log("Shop Item Details: " + shop.shopItemDetails);
+
+  const detailedShopItem = shop.shopItemDetails.find(
```
Comment (crypt0s1s) [inline Helios/pages/shop/[id].tsx:16]: again I think you misunderstood that the shopItemDetails call only returns a single object

File: Selene/.env
Comment (crypt0s1s) [inline Selene/.env:1]: Thanks for doing this

File: Helios/api/store/shop/store.ts (hunk 1 above)
Comment (MitchMunro) [reply Helios/api/store/shop/store.ts:7]: Removed.

File: Helios/api/store/shop/store.ts (hunk 3 above)
Comment (MitchMunro) [reply Helios/api/store/shop/store.ts]: Shall do

File: Helios/pages/shop/index.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        </div>
+      </div>
+    </main>
+  );
+};
+
+const ShopItemGrid = observer(() => {
+  const { shop } = useContext(StoreContext);
+  const { isPending, isSuccess, error } = getShop();
+  console.log(error);
```
[the remark this answers is not in the document]
Comment (MitchMunro) [reply Helios/pages/shop/index.tsx]: I removed them too

File: Selene/.env [hunk 10]
```diff
@@ -1,6 +1,6 @@
 SECRET=JOKER
 DATABASE_HOSTNAME="localhost"
 DATABASE_PORT=5432
-DATABASE_USERNAME="joshuasumskas"
+DATABASE_USERNAME="Mitch"
```
[the remark this answers is not in the document]
Comment (MitchMunro) [reply Selene/.env]: nice nice

**Review by MitchMunro — COMMENTED** [review]
Comments taken into account and changed

**Pull request merged**